### ISSUE TITLE
[sklearn#34312] `CalibratedClassifierCV(method="sigmoid")` should always take the raw logits as inputs instead of the predicted probabilities

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#337]
**Mirrored from [scikit-learn#34312](https://github.com/scikit-learn/scikit-learn/issues/34312)**

Original title: `CalibratedClassifierCV(method="sigmoid")` should always take the raw logits as inputs instead of the predicted probabilities
Original labels: Bug, module:calibration
Created: 2026-06-16T13:57:08Z

---

This is a defect [I first observed while reviewing the temperature scaling PR](https://github.com/scikit-learn/scikit-learn/pull/31068#discussion_r2012459507): if the uncalibrated classifier only exposes a `decision_function` and no `predict_proba` (which is the case for SVMs for instance) our implementation of sigmoid calibration (aka Platt-scaling) is fine because the output values of `decision_function` are morally logits between -inf and +inf with 0 as the default decision threshold.

But if the uncalibrated classifier already exposes a `predict_proba` method (e.g. `GaussianNB`), our current implementation takes the predicted probabilities as input of the parametrized sigmoid function. This is incorrect: we should the log of the probabilistic predictions (incremented by a small epsilon) to feed logits as inputs to the sigmoid function.

I had started an unfinished WIP branch now turned into a draft PR (https://github.com/scikit-learn/scikit-learn/pull/34313) to attempt to fix this problem and started to do empirical evaluation to assess the impact of this change in examples. Early results were very promising. However, it would now need work to be sync with main again after the merge of the temperature scaling PR and I got distracted and this was burried in my personal TODO list of WIP stuff to revive.

Meanwhile, the following paper by @eugeneberta, @dholzmueller et al. was published, and it confirms that it is empirically much better to take the logits, in which case it is a very competitive calibration method (better than most other calibration methods on average on a large set of data & model pairs).

CalArena: A Large-Scale Post-Hoc Calibration Benchmark
https://arxiv.org/abs/2605.30188

<img width="3740" height="1463" alt="Image" src="https://github.com/user-attachments/assets/72703305-b5ff-4fb2-8956-4441640ec3a2" />

<img width="3745" height="1463" alt="Image" src="https://github.com/user-attachments/assets/da526b6e-7ca5-4c03-811b-4d8d3b560a34" />

Side note: I suspect that OvR Platt-logits followed by post-hoc class-wise renormalization would also work very well in the multiclass setting, but this does not seem to have been explored in the above paper from a cursory glance. Do you confirm @eugeneberta, @dholzmueller?

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/86 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
